transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java

```diff
@@ -807,6 +807,14 @@ private void write(Object msg, boolean flush, ChannelPromise promise) {
                 task = WriteTask.newInstance(next, m, promise);
             }
             safeExecute(executor, task, promise, m);
+            if (!flush) {
+                /**
+                 * Whenever there is a write, if auto-flush is enabled, the selector is required to be woken up,
+                 * otherwise the write may never be flushed or have latency.
+                 * This method just wakes up the selector once till the next auto-flush task is run.
```

Comment:
maybe replace selector with eventloop here ? As a selector not exists for all impls. This will be more correct ;) Just a nit...